build.ps1

```diff
@@ -1,12 +1,32 @@
 
+$ErrorActionPreference = "Stop"
 $VERSION="dev"
 $SHA=$(git rev-parse --verify HEAD)
 $BUILDDATE=Get-Date -Format "yyyy/MM/dd HH:mm:ss zzz"
 $GOVERSION=$(go version)
 $LDFLAGS="-X 'main.version=${VERSION}' -X 'main.sha=${SHA}' -X 'main.builddate=${BUILDDATE}' -X 'main.goversion=${GOVERSION}'"
 
 Invoke-Expression "$(Join-Path ${env:GOPATH}\bin govendor.exe -Resolve) sync"
-Invoke-Expression "$(Join-Path ${env:GOPATH}\bin go-bindata.exe -Resolve) -pkg registry -o registry\schema.bindata.go schema"
+$modtime=$(git log -n 1 --date=raw --pretty=format:%cd -- schema/).split(" ", 2)[0]
+Invoke-Expression "$(Join-Path ${env:GOPATH}\bin go-bindata.exe -Resolve) -mode 420 -modtime ${modtime} -pkg registry -o registry\schema.bindata.go schema"
+
+# Determine the default branch reference for registry/github.go
+$SCHEMA_LAST_COMMIT=$(git log -n 1 --pretty=format:%H -- schema/ registry/schema.bindata.go)
+$f=$(git rev-list origin/master | Select-String "${SCHEMA_LAST_COMMIT}" -ErrorAction "Continue")
+if ( $f -ne $null ) {
+  # Found no changes for resource template/schema on HEAD.
+  # so that set preference to the master branch.
+  $LDFLAGS="${LDFLAGS} -X 'registry.GithubDefaultRef=master'"
+} else {
+  # Found resource template/schema changes on this HEAD. Switch the default reference branch.
+  # Check if $GIT_BRANCH has something once in case of running in Jenkins.
+  $branch = if( $env:GIT_BRANCH ){
+      $env:GIT_BRANCH
+  }else{
+      $(git rev-parse --abbrev-ref HEAD)
+  }
+  $LDFLAGS="${LDFLAGS} -X 'registry.GithubDefaultRef=${branch}'"
+}

```

Comment:
The functionality is to work the CLI syntax following:

```
openvdc run centos/7/lxc
```

The the template file is assumed to be on remote server in this case and it is expected to be on github as default. During development, the "templates/" folder will be changed so we want to set the default branch to a topic branch. For release build, it should be ok with the master branch.